### PR TITLE
Replace unregistered token icon with identicon

### PR DESCRIPTION
**Pull request by BMillman19**

### CODE REVIEW

**Comment by BMillman19** [comment]
I realized theres a bunch of places i missed

**Comment by fabioberger** [comment]
Nice! This is great but can we also make sure that the identicon is shown on the "Trade History" page and the "Fill order" order preview?

**Comment by BMillman19** [comment]
Places I checked for identicon usage:

- Generate order
- Asset picker
- Fill order
- Balances

File: ts/components/ui/token_icon.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    diameter: number;
+}
+
+interface TokenIconState {}
+
+export class TokenIcon extends React.Component<TokenIconProps, TokenIconState> {
+    public render() {
+        return (
+            <div>
+                {(this.props.token.isRegistered) ?
```
Comment (fabioberger) [inline ts/components/ui/token_icon.tsx]: Nit: unnecessary parens.

**Comment by fabioberger** [comment]
This looks great! The identicons look a bit big compared to the logo'd tokens despite being the same dimensions. Could we try with scaling the identicon down by 10%? If it looks good then this is ready to go.

Let's wait for a "go ahead" from the rest of the team before merging.

File: ts/components/ui/token_icon.tsx (hunk 1 above)
Comment (BMillman19) [reply ts/components/ui/token_icon.tsx]: this is a convention we used at twitter, it does make the expression easier to read when it is a one liner. for example let x = (y < 1) ? someVar : otherVar

**Comment by BMillman19** [comment]
Regarding identicon sizing, it may be easier for you to fine tune with Ben by your side, the layout logic may need some updating because when the identicon diameter is smaller, the text for the address and "registered token" line is too far up compared to the token logo in the fill order view

**Comment by fabioberger** [comment]
Regression:

Filling an order with a custom, unverified token results in:

screen shot 2017-10-05 at 17 33 43


Sample order on Kovan:

```
{"maker":{"address":"0x198c6ad858f213fb31b6fe809e25040e6b964593","token":{"name":"Maker DAO","symbol":"MKR","decimals":18,"address":"0x1dad4783cf3fe3085c1426157ab175a6119a04ba"},"amount":"4000000000000000000","feeAmount":"0"},"taker":{"address":"","token":{"name":"Custom Token","symbol":"CUS","decimals":18,"address":"0x63d26819c90de69f6e5059fb1809c6bf04f24005"},"amount":"1000000000000000000","feeAmount":"0"},"expiration":"2524608000","feeRecipient":"0x0000000000000000000000000000000000000000","salt":"68975638728266675041929907546211807313389096844188107422166495153556489514450","signature":{"v":28,"r":"0xe82544d43ceb3161fe4753280d2d25f5253342f59895e46f0c50997bcf01e649","s":"0x059e249c8b85d64c84d68474e510366026dd7d490f41d99aa1880a30d0180499","hash":"0x3ee69fd4f8a4c79fa53931dbee8aacaaa66e6f45cf580f29014448091a83aec6"},"exchangeContract":"0x90fe2af704b34e0224bf2299c838e04d4dcf1364","networkId":42}
```

**Comment by fabioberger** [comment]
LGTM

**Pull request merged**